src/crewai/cli/deploy/api.py

```diff
@@ -0,0 +1,63 @@
+from os import getenv
+
+import requests
+
+
+class CrewAPI:
+    """
+    CrewAPI class to interact with the crewAI+ API.
+    """
+
+    def __init__(self, api_key: str) -> None:
+        self.api_key = api_key
+        self.headers = {
+            "Authorization": f"Bearer {api_key}",
+            "Content-Type": "application/json",
+        }
+        self.base_url = getenv(
+            "CREWAI_BASE_URL", "https://dev.crewai.com/crewai_plus/api/v1/crews"
+        )
+
+    def _make_request(self, method: str, endpoint: str, **kwargs) -> requests.Response:
+        url = f"{self.base_url}/{endpoint}"
+        return requests.request(method, url, headers=self.headers, **kwargs)
```

Comment:
Also no need to change: I've been using Session Objects in such cases. I love how I can setup the object then simply call `get`, `post`, and so on without passing auth headers, url and other stuff.

Reply:
Awesome, I'll take into account for next iterations 🫡